lib/rucio/core/replica.py

```diff
@@ -468,18 +492,25 @@ def get_pfn_to_rse(pfns, vo='def', *, session: "Session"):
         if storage_element not in storage_elements:
             storage_elements.append(storage_element)
             se_condition.append(models.RSEProtocols.hostname == storage_element)
-    query = session.query(models.RSEProtocols.rse_id,
-                          models.RSEProtocols.scheme,
-                          models.RSEProtocols.hostname,
-                          models.RSEProtocols.port,
-                          models.RSEProtocols.prefix).\
-        join(models.RSE, models.RSE.id == models.RSEProtocols.rse_id).\
-        filter(and_(or_(*se_condition), models.RSEProtocols.scheme == scheme)).\
-        filter(models.RSE.deleted == false()).\
-        filter(models.RSE.staging_area == false())
+    stmt = select(
+        models.RSEProtocols.rse_id,
+        models.RSEProtocols.scheme,
+        models.RSEProtocols.hostname,
+        models.RSEProtocols.port,
+        models.RSEProtocols.prefix
+    ).join(
+        models.RSE, models.RSE.id == models.RSEProtocols.rse_id
+    ).where(
+        and_(or_(*se_condition),
+             models.RSEProtocols.scheme == scheme)
+    ).where(
+        and_(models.RSE.deleted == false(),
+             models.RSE.staging_area == false())
+    )
```

Comment:
I think this could also be written under a single `where` statement, but it would functionally be the same so I don't think it's necessary to make a change